lib/features/email/domain/usecases/get_email_content_interactor.dart

```diff
@@ -28,37 +31,85 @@ class GetEmailContentInteractor {
-          baseDownloadUrl,
-          accountId,
-          draftsEmail: draftsEmail
-        );
+      if (!BuildUtils.isWeb) {
+        yield* _getOpenedEmailCache(session, accountId, emailId, baseDownloadUrl, composeEmail: composeEmail, draftsEmail: draftsEmail);
+      } else {
+        yield* _getContentEmailFromServer(session, accountId, emailId, baseDownloadUrl, composeEmail: composeEmail, draftsEmail: draftsEmail);
+      }
+    } catch (e) {
+      log('GetEmailContentInteractor::execute(): exception = $e');
+      yield Left(GetEmailContentFailure(e));
+    }
+  }
 
-        final newEmailContentsDisplayed = BuildUtils.isWeb && !composeEmail
-          ? await emailRepository.addTooltipWhenHoverOnLink(newEmailContents)
-          : newEmailContents;
+  Stream<Either<Failure, Success>> _getContentEmailFromServer(
+    Session session,
+    AccountId accountId,
+    EmailId emailId,
+    String? baseDownloadUrl,
+    {
+      bool composeEmail = false,
+      bool draftsEmail = false
+    }
+  ) async* {
+    final email = await emailRepository.getEmailContent(session, accountId, emailId);
 
-        yield Right<Failure, Success>(GetEmailContentSuccess(
-          newEmailContents,
-          newEmailContentsDisplayed,
-          email.allAttachments,
-          email
+    if (email.emailContentList.isNotEmpty) {
+      final newEmailContents = await emailRepository.transformEmailContent(
+        email.emailContentList,
+        email.allAttachments.listAttachmentsDisplayedInContent,
+        baseDownloadUrl,
+        accountId,
+        draftsEmail: draftsEmail
+      );
+
+      final newEmailContentsDisplayed = BuildUtils.isWeb && !composeEmail
+        ? await emailRepository.addTooltipWhenHoverOnLink(newEmailContents)
+        : newEmailContents;
+
+      yield Right<Failure, Success>(GetEmailContentSuccess(
+        newEmailContents,
+        newEmailContentsDisplayed,
+        email.allAttachments,
+        email
+      ));
+    } else if (email.allAttachments.isNotEmpty) {
+      yield Right<Failure, Success>(GetEmailContentSuccess([], [], email.allAttachments, email));
+    } else if (email.headers?.isNotEmpty == true) {
+      yield Right<Failure, Success>(GetEmailContentSuccess([], [], [], email));
+    } else {
+      yield Left(GetEmailContentFailure(null));
+    }
+  }
+
+  Stream<Either<Failure, Success>> _getOpenedEmailCache(
+    Session session,
+    AccountId accountId,
+    EmailId emailId,
+    String? baseDownloadUrl,
+      {
+        bool composeEmail = false,
+        bool draftsEmail = false
+      }
+  ) async* {
+    log('GetEmailContentInteractor::_getOpenedEmailCache():');
+    try {
+      final detailedEmail = await emailRepository.getOpenedEmail(session, accountId, emailId);
+      if (detailedEmail != null) {
+        yield Right<Failure, Success>(GetEmailContentFromCacheSuccess(
+          detailedEmail.htmlEmailContent!,
+          detailedEmail.attachments!,
```

Comment:
Notice the case one of the 3 values (`htmlEmailContent, attachments, header`) can be null but the email is still displayed